src/Arcus.Testing.Tests.Integration/Integration/DataFactory/Fixture/TemporaryDataFactoryDataFlow.cs

```diff
@@ -200,27 +301,51 @@ private async Task AddCsvSinkAsync(AssertCsvOptions options)
                 DataLocation = new AzureBlobStorageLocation
                 {
                     Container = _sourceContainer?.Name ?? throw new InvalidOperationException("Azure blob storage container should be available at this point"),
-                    FolderPath = _sinkDatasetName
+                    FolderPath = SinkDataSetName
                 }
             };
 
             await _sinkDataset.UpdateAsync(WaitUntil.Completed, new DataFactoryDatasetData(sinkProperties));
         }
 
+        private async Task AddCsvSinkAsyncWithDataSetParameterAsync(AssertCsvOptions options)
+        {
+            _logger.LogTrace("Adding CSV sink '{SinkName}' to DataFlow '{DataFlowName}' within Azure DataFactory '{DataFactoryName}'", SinkDataSetName, Name, DataFactory.Name);
+
+            var blobStorageLinkedService = new DataFactoryLinkedServiceReference(DataFactoryLinkedServiceReferenceKind.LinkedServiceReference, _linkedServiceName);
+
+            _sinkDataset = _arm.GetDataFactoryDatasetResource(DataFactoryDatasetResource.CreateResourceIdentifier(SubscriptionId, ResourceGroupName, DataFactory.Name, SinkDataSetName));
+            var sinkProperties = new DelimitedTextDataset(blobStorageLinkedService)
+            {
+                ColumnDelimiter = options.Separator.ToString(),
+                RowDelimiter = options.NewLine,
+                QuoteChar = options.Quote.ToString(),
+                FirstRowAsHeader = options.Header is AssertCsvHeader.Present,
+                DataLocation = new AzureBlobStorageLocation
+                {
+                    Container = _sourceContainer?.Name ?? throw new InvalidOperationException("Azure blob storage container should be available at this point"),
+                    FolderPath = DataFactoryElement<string>.FromExpression($"@concat('{SinkDataSetName}/', dataset().{SinkDataSetParameterKey})")
```

Comment:
This line is the only difference to make it parameterized, right? Then this should be available/configurable to the test: less duplication + clear from the test body what gets parameterized.

Something like:
```csharp
public class TempDataFlowOptions
{
    public TempDataFlowSinkOptions Sink { get; } = new();
}

public class TempDataFlowSinkOptions
{
    private readonly Collection<string> _subPaths;

    public TempDataFlowSinkOptions AddFolderSubPath(string subPath)
    {
        _subPaths.Add(subPath);
    }

    internal void ApplyOptions(DelimitedTextDataset dataSet)
    {
        if (_subPaths.Count > 1)
        {
            dataSet.FolderPath = DataFactoryElement<string>.FromExpression($"@concat('{SinkDataSetName}/', dataset().{string.Join(".", _subPaths)})");
        }
    }
}
```

Reply:
Applied change in [#ed1001a](https://github.com/arcus-azure/arcus.testing/pull/228/commits/ed1001a1f78508ff48b30b4ab5a711c48bbd3c33)